src/components/board/board.test.js

```diff
@@ -1,8 +1,37 @@
+const board = require('../../services/boardService');
 
+const FILES = ['A', 'B', 'C', 'D', 'E', 'F', 'G', 'H'];
+const RANKS = [1, 2, 3, 4, 5, 6, 7, 8];
 
+describe('board.js', () => {
+  // testing to make sure the array is populated with 64 object values
+  test('Function will create an array of square objects', () => {
+    expect (board(FILES, RANKS)).toHaveLength(64);
+  });
 
-it('renders without crashing', () => {
-  let value = true;
-  expect(value).toBe(true);
+  // testing to make sure thhe objects have the required grid properties
+  test('Function will have objects with chessboard grid property values ', () => {
+    expect(board(FILES, RANKS)).toEqual(
+      expect.arrayContaining([
+        expect.objectContaining({
+          coordinate: '1A',
```

Comment:
Sense you're trying to test a random square name exists, try using something that will select a random value from FILES and a random value from RANKS and just have one test.

Something like this - not tested though.

const FILES = ['A', 'B', 'C', 'D', 'E', 'F', 'G', 'H'];
const RANKS = [1, 2, 3, 4, 5, 6, 7, 8];
const randomFiles = FILES[Math.floor(Math.random() * FILES.length)];
const randomRanks = RANKS[Math.floor(Math.random() * RANKS.length)];
const randomdFileRank = `${randomFiles}${randomRanks}`

Reply:
oh that looks great, didnt think of that!